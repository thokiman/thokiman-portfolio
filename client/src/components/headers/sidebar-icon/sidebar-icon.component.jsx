import useWindowDimensions from 'components/hooks/window-dimensions/useWindowDimensions.component.jsx';
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { toggleSideBarActive } from '../../../redux/header/header.actions.js';
import { selectIsSideBarActive } from '../../../redux/header/header.selectors';
import {
  SideBarHeaderIcon,
  SideBarOne,
  SideBarThree,
  SideBarTwo,
} from './sidebar-icon.styles';
import {
  measureSideBarOneStyles,
  measureSideBarThreeStyles,
} from './sidebar-icon.utils.style.js';

export const SideBarIcon = ({
  toggleSideBarActive,
  isSideBarHidden: isSideBarActive,
}) => {
  const { viewWidth, viewHeight } = useWindowDimensions();
  return (
    <SideBarHeaderIcon
      $issidebaractive={isSideBarActive}
      onClick={() => toggleSideBarActive()}
    >
      <SideBarOne
        $measuresidebarone={measureSideBarOneStyles(viewWidth, viewHeight)}
        $issidebaractive={isSideBarActive}
      />
      <SideBarTwo $issidebaractive={isSideBarActive} />
      <SideBarThree
        $measuresidebarthree={measureSideBarThreeStyles(viewWidth, viewHeight)}
        $issidebaractive={isSideBarActive}
      />
    </SideBarHeaderIcon>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarActive,
});
const mapDispatchToProps = (dispatch) => ({
  toggleSideBarActive: () => dispatch(toggleSideBarActive()),
});

export default connect(mapStateToProps, mapDispatchToProps)(SideBarIcon);
