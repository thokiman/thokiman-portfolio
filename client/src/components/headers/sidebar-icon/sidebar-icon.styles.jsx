import styled, { css, keyframes } from 'styled-components';

const greenBlueColor = '#128194';
const maxWidthSmartphone = 800;
//bar
const SideBarHeaderIconIsActived = css`
  top: 50px;
  left: 50px;
`;
const getSideBarHeaderIconActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return SideBarHeaderIconIsActived;

  return null;
};

export const SideBarHeaderIcon = styled.div`
  display: none;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    display: block;
    position: absolute;
    z-index: 1;
    top: 5vh;
    left: 10vw;
    cursor: pointer;
    width: fit-content;
    height: fit-content;

    ${getSideBarHeaderIconActive}
  }
`;

const SideBarHideStyles = css`
  width: 9vw;
  height: 0.4vh;
  background-color: ${greenBlueColor};
  margin: 6px 0;
  transition: 0.6s;
  position: relative;
  top: 10px;
  left: -15px;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 6vw;
  }
  @media only screen and (max-width: 550px) {
    width: 8vw;
    top: 0px;
    left: 0px;
  }
  @media only screen and (max-width: 400px) {
    width: 9vw;
  }
  @media only screen and (max-width: 350px) {
    width: 12vw;
  }
`;

const SideBarOneHide = css`
  ${SideBarHideStyles}
`;

const SideBarTwoHide = css`
  ${SideBarHideStyles}
`;

const SideBarThreeHide = css`
  ${SideBarHideStyles}
`;

//bar change
const sideBarOneChange = keyframes`
    from {
      transform: rotate(0deg) ;
    }

    to {
      transform: rotate(-45deg) ;
    }
`;

const sideBarThreeChange = keyframes`
    from {
      transform: rotate(0deg) ;
    }

    to {
      transform: rotate(45deg) ;
    }
`;
const SideBarChangeStyles = css`
  width: 9vw;
  height: 0.4vh;
  background-color: ${greenBlueColor};
  border-radius: 20px 20px 20px 20px;
`;

const SideBarOneChange = css`
  ${SideBarChangeStyles}
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    top: 20px;
    left: -20px;
    transform: rotate(-45deg);
    animation: ${sideBarOneChange} 0.4s ease-in-out;
  }
  @media only screen and (max-width: 550px) {
    top: 0px;
    left: -9px;
  }

  @media only screen and (max-width: 300px) {
    top: 7px;
    left: 3px;
  }
`;

const SideBarTwoChange = css`
  opacity: 0;
`;

const SideBarThreeChange = css`
  ${SideBarChangeStyles}
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    top: 18px;
    left: -15px;
    transform: rotate(45deg);
    animation: ${sideBarThreeChange} 0.4s ease-in-out;
  }
  @media only screen and (max-width: 550px) {
    top: -3px;
    left: -5px;
  }
  @media only screen and (max-width: 350px) {
    top: -1px;
  }
  @media only screen and (max-width: 300px) {
    top: 5px;
    left: 5px;
  }
`;

const getSideBarOne = ({ $issidebaractive }) => {
  if ($issidebaractive) {
    return SideBarOneChange;
  }
  return SideBarOneHide;
};
const getSideBarTwo = ({ $issidebaractive }) => {
  if ($issidebaractive) {
    return SideBarTwoChange;
  }
  return SideBarTwoHide;
};
const getSideBarThree = ({ $issidebaractive }) => {
  if ($issidebaractive) {
    return SideBarThreeChange;
  }
  return SideBarThreeHide;
};

export const SideBarOne = styled.div`
  ${getSideBarOne}
`;
export const SideBarTwo = styled.div`
  ${getSideBarTwo}
`;
export const SideBarThree = styled.div`
  ${getSideBarThree}
`;
SideBarOne.displayName = 'SideBarOne';
SideBarTwo.displayName = 'SideBarTwo';
SideBarThree.displayName = 'SideBarThree';
SideBarHeaderIcon.displayName = 'SideBarHeaderIcon';
