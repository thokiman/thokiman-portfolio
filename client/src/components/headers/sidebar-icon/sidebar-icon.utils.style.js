import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../utils/common.component.utils';

export const measureSideBarOneStyles = (viewWidth, viewHeight) => {
  let sideBarOneProps;
  //left

  if (hasScreenSize1(viewWidth)) {
    sideBarOneProps = {
      top: 2.1,
      left: -2.5,
    };
  } else if (hasScreenSize2(viewWidth)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  } else if (hasScreenSize5(viewWidth)) {
    sideBarOneProps = {
      top: 1,
      left: 1,
    };
  } else {
    sideBarOneProps = {
      top: 0.1,
      left: -1.4,
    };
  }
  return sideBarOneProps;
};

export const measureSideBarThreeStyles = (viewWidth, viewHeight) => {
  // right
  let sideBarThreeProps;

  if (hasScreenSize1(viewWidth)) {
    sideBarThreeProps = {
      top: 1.75,
      left: -1.6,
    };
  } else if (hasScreenSize2(viewWidth)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    sideBarThreeProps = {
      top: -0.3,
      left: -0.6,
    };
  } else if (hasScreenSize5(viewWidth)) {
    sideBarThreeProps = {
      top: 0.7,
      left: 1.9,
    };
  } else {
    sideBarThreeProps = {
      top: -0.2,
      left: -0.4,
    };
  }
  return sideBarThreeProps;
};
