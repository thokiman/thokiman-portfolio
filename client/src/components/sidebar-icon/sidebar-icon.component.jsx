import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
import { toggleSideBarHidden } from "../../redux/header/header.actions.js";
import {
  SideBarOne,
  SideBarThree,
  SideBarTwo,
  SideBarHeaderIcon,
} from "./sidebar-icon.styles";

export const SideBarIcon = ({ toggleSideBarHidden, isSideBarHidden }) => {
  return (
    <SideBarHeaderIcon
      $issidebarhidden={isSideBarHidden}
      onClick={() => toggleSideBarHidden()}
    >
      <SideBarOne $issidebarhidden={isSideBarHidden} />
      <SideBarTwo $issidebarhidden={isSideBarHidden} />
      <SideBarThree $issidebarhidden={isSideBarHidden} />
    </SideBarHeaderIcon>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
});
const mapDispatchToProps = (dispatch) => ({
  toggleSideBarHidden: () => dispatch(toggleSideBarHidden()),
});

export default connect(mapStateToProps, mapDispatchToProps)(SideBarIcon);
