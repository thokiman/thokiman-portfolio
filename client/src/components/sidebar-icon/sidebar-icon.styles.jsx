import styled, { css, keyframes } from "styled-components";

//bar

const SideBarHeaderIconActive = css`
  top: 50px;
  left: 50px;
`;
const getSideBarHeaderIconActive = (props) => {
  if (props.$issidebarhidden) return SideBarHeaderIconActive;
  return null;
};

export const SideBarHeaderIcon = styled.div`
  display: none;
  @media only screen and (max-width: 800px) {
    display: block;
    position: absolute;
    z-index: 1;
    top: 35px;
    left: 49px;
    cursor: pointer;
    width: fit-content;
    height: fit-content;
    ${getSideBarHeaderIconActive}
  }
`;

const SideBarHideStyles = css`
  width: 35px;
  height: 3px;
  background-color: #128194;
  margin: 6px 0;
  transition: 0.6s;
`;

const SideBarOneHide = css`
  ${SideBarHideStyles}
`;

const SideBarTwoHide = css`
  ${SideBarHideStyles}
`;

const SideBarThreeHide = css`
  ${SideBarHideStyles}
`;

//bar change
const sideBarOneChange = keyframes`
    from {
      transform: rotate(0deg) translate(0px, 0px);
    }

    to {
      transform: rotate(-45deg) translate(-3px, -1px);
    }
`;

const sideBarThreeChange = keyframes`
    from {
      transform: rotate(0deg) translate(0px, 0px);
    }

    to {
      transform: rotate(45deg) translate(-2px, -1px);
    }
`;
const SideBarChangeStyles = css`
  width: 30px;
  height: 3px;
  background-color: #128194;
  border-radius: 20px 20px 20px 20px;
`;

const SideBarOneChange = css`
  ${SideBarChangeStyles}
  transform: rotate(-45deg) translate(-3px, -1px);
  animation: ${sideBarOneChange} 0.4s ease-in-out;
`;

const SideBarTwoChange = css`
  opacity: 0;
`;

const SideBarThreeChange = css`
  ${SideBarChangeStyles}

  transform: rotate(45deg) translate(-2px, -1px);
  animation: ${sideBarThreeChange} 0.4s ease-in-out;
`;

const getSideBarOne = ({ $issidebarhidden }) => {
  if ($issidebarhidden) {
    return SideBarOneChange;
  }
  return SideBarOneHide;
};
const getSideBarTwo = ({ $issidebarhidden }) => {
  if ($issidebarhidden) {
    return SideBarTwoChange;
  }
  return SideBarTwoHide;
};
const getSideBarThree = ({ $issidebarhidden }) => {
  if ($issidebarhidden) {
    return SideBarThreeChange;
  }
  return SideBarThreeHide;
};

export const SideBarOne = styled.div`
  ${getSideBarOne}
`;
export const SideBarTwo = styled.div`
  ${getSideBarTwo}
`;
export const SideBarThree = styled.div`
  ${getSideBarThree}
`;
SideBarOne.displayName = "SideBarOne";
SideBarTwo.displayName = "SideBarTwo";
SideBarThree.displayName = "SideBarThree";
SideBarHeaderIcon.displayName = "SideBarHeaderIcon";
