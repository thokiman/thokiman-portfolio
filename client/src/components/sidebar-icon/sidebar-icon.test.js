import React from "react";
import { shallow } from "enzyme";

import { SideBarIcon } from "./sidebar-icon.component";

describe("SideBarIcon component", () => {
  let wrapper;
  let mockToggleSideBarHidden;
  const mockFalseBool = false;

  beforeEach(() => {
    mockToggleSideBarHidden = jest.fn();
    const mockProps = {
      toggleSideBarHidden: mockToggleSideBarHidden,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<SideBarIcon {...mockProps} />);
  });

  it("should render SideBarIcon component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
