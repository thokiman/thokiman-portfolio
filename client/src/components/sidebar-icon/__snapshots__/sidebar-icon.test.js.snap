// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarIcon component should render SideBarIcon component 1`] = `ShallowWrapper {}`;
